content/sensu-core/2.0/reference/hooks.md

```diff
@@ -0,0 +1,96 @@
+---
+title: "Hooks"
+description: "The hooks reference guide."
+weight: 1
+version: "2.0"
+product: "Sensu Core"
+platformContent: true
+menu:
+  sensu-core-2.0:
+    parent: reference
+---
+
+# Hooks
+
+### What are hooks?
+
+Hooks are commands run by the Sensu client in response to the result of a check, mutator, or handler execution. The Sensu client will execute the appropriate configured hook command, depending on the execution status (e.g. 1). The hook command output, status, executed timestamp, and duration are captured and published in the event result. Hook commands can optionally receive JSON serialized Sensu client data via STDIN.
```

Comment:
> in response to the result of a check, mutator, or handler execution

I might be wrong but I believe hooks are only run in response to the result of a check command execution right now; I don't see any reference to hooks in the handler & mutator structs!

Reply:
Yes its not implemented yet! Do you think it would be reasonable to reference the feature tickets associated with mutator and handler hooks, or just remove that part completely?

Reply:
IMO we should remove references to things that are unimplemented, but create a TODO in the feature tickets that links back to this document for including them once they exist.

Reply:
The reusability of hooks _allows_ us to add them to mutator and handlers. Otherwise, what's so different about 2.x hooks and why are we differentiating "hooks" and "check hooks." Just thinking out loud.. do you think the documentation is clear enough about that difference if we were to remove any references to mutator or handlers?

Reply:
We could definitively add a mention in the `New and improved hooks!` section mentioning that since hooks are now a distinct resource, it opens the door for their use with other type of resources than check in the future. Not mentioning _what_ resources and _when_ might help managing expectations!